### PR TITLE
test-utils: add `data/`

### PR DESCRIPTION
**Pull request by hinto-janai**
- Adds a `cuprate_test_utils::data` module containing raw and typed Monero data for testing.

This was a `#[cfg(test)]` submodule in `database/` but I thought it would make more sense here.

(Also adds lints and `clippy --fix`es a bunch of things)

### CODE REVIEW

File: test-utils/src/monerod/download.rs [hunk 1]
```diff
@@ -24,23 +24,23 @@ static DOWNLOAD_MONEROD_MUTEX: Mutex<()> = Mutex::const_new(());
...
-        ("windows", "x64") | ("windows", "x86_64") => format!("monero-win-x64-{}.zip", version),
-        ("windows", "x86") => format!("monero-win-x86-{}.zip", version),
-        ("linux", "x64") | ("linux", "x86_64") => format!("monero-linux-x64-{}.tar.bz2", version),
-        ("linux", "x86") => format!("monero-linux-x86-{}.tar.bz2", version),
-        ("macos", "x64") | ("macos", "x86_64") => format!("monero-mac-x64-{}.tar.bz2", version),
+        ("windows", "x64" | "x86_64") => format!("monero-win-x64-{version}.zip"),
+        ("windows", "x86") => format!("monero-win-x86-{version}.zip"),
+        ("linux", "x64" | "x86_64") => format!("monero-linux-x64-{version}.tar.bz2"),
+        ("linux", "x86") => format!("monero-linux-x86-{version}.tar.bz2"),
+        ("macos", "x64" | "x86_64") => format!("monero-mac-x64-{version}.tar.bz2"),
```
Comment (hinto-janai) [inline test-utils/src/monerod/download.rs:31]: This is turning:
```
("windows", "x64") | ("windows", "x86_64")
```
into:
```
("windows", "x64" | "x86_64")
```
I didn't know I could like Rust's pattern matching even more :D

**Review by hinto-janai — COMMENTED** [review]
@Boog900 ready - https://github.com/Cuprate/cuprate/pull/102 is waiting on this btw.

File: test-utils/src/monerod/download.rs (hunk 1 above)
Comment (Boog900) [reply test-utils/src/monerod/download.rs:31]: Wow yeah TIL

**Comment by hinto-janai** [comment]
Actually don't merge yet - need to add more `Block` data.

**Comment by hinto-janai** [comment]
Okay ready.

File: test-utils/src/monerod.rs [hunk 2]
```diff
@@ -17,16 +17,23 @@ use tokio::{task::yield_now, time::timeout};
...
 const MONEROD_STARTUP_TEXT: &str =
     "The daemon will start synchronizing with the network. This may take a long time to complete.";
 
+/// The log line `monerod` emits indicated it has stopped.
 const MONEROD_SHUTDOWN_TEXT: &str = "Stopping cryptonote protocol";
 
 /// Spawns monerod and returns [`SpawnedMoneroD`].
 ///
 /// This function will set `regtest` and the P2P/ RPC ports so these can't be included in the flags.
+#[allow(clippy::future_not_send)]
```
Comment (Boog900) [inline test-utils/src/monerod.rs:36]: I dont think this allow is needed anymore

File: test-utils/src/monerod.rs [hunk 3]
```diff
@@ -159,6 +165,7 @@ impl Drop for SpawnedMoneroD {
             println!("------END-MONEROD-LOGS------");
         }
 
+        #[allow(clippy::manual_assert)] // clippy suggests some real unreadable things sometimes
```
Comment (Boog900) [inline test-utils/src/monerod.rs:168]: or this one

File: test-utils/src/monerod/download.rs [hunk 4]
```diff
@@ -83,7 +83,8 @@ fn find_target() -> PathBuf {
 }
 
 /// Checks if we have monerod or downloads it if we don't and then returns the path to it.
-pub async fn check_download_monerod() -> Result<PathBuf, ReqError> {
+#[allow(clippy::redundant_pub_crate)]
```
Comment (Boog900) [inline test-utils/src/monerod/download.rs:86]: and here :)

**Review by Boog900 — APPROVED** [review]

**Pull request merged**